src/d3a/models/area/stats.py

```diff
@@ -90,3 +95,39 @@ def cheapest_offers(self):
         for market in self._markets.markets.values():
             cheapest_offers.extend(market.sorted_offers[0:1])
         return cheapest_offers
+
+    def min_max_avg_rate_market(self, time_slot):
+        out_dict = {"min_trade_rate": None,
+                    "max_trade_rate": None,
+                    "avg_trade_rate": None,
+                    "total_traded_energy_kWh": None}
+        for market in self._markets.all_spot_markets:
+            if market.time_slot == time_slot and len(market.trades) > 0:
+                trade_volumes = [trade.offer.energy for trade in market.trades]
+                trade_rates = [trade.offer.price/trade.offer.energy for trade in market.trades]
+                out_dict["min_trade_rate"] = limit_float_precision(min(trade_rates))
+                out_dict["max_trade_rate"] = limit_float_precision(max(trade_rates))
+                out_dict["avg_trade_rate"] = limit_float_precision(mean(trade_rates))
+                out_dict["total_traded_energy_kWh"] = limit_float_precision(sum(trade_volumes))
+        return out_dict
+
+    @property
+    def current_market(self):
+        past_markets = list(self._markets.past_markets.values())
+        return past_markets[-1] if len(past_markets) > 0 else None
+
+    def get_market_price_stats(self, market_slot_list):
+        if self.current_market is None:
+            return {"INFO": "No market stats available yet"}
```

Comment:
I would prefer if, instead of info, there was the source of the message, `market_stats`

Reply:
in RedisMarketExternalConnection.market_stats_callback the ret val of this function is already assigne tot he member "market_stats". IMO your proposal would duplicate this. 
But i am happy about other renaming proposals.